### ISSUE TITLE
Upgrade from Debian Bullseye to Bookworm

### ISSUE THREAD

**Issue by markmandel** [googleforgames/agones#3570]
Just noticing we're using Debian 11 everywhere. It's currently being kept up to date with security fixes, but it would be good to be proactive and update to Debian 12.

This can definitely be done as a series of smaller PR's.

For:
https://github.com/search?q=repo%3Agoogleforgames%2Fagones%20debian%3Abullseye&type=code
Switch from `debian:bullseye` to `debian:bookworm`

For:
https://github.com/search?q=repo%3Agoogleforgames%2Fagones%20debian11&type=code
Switch from `gcr.io/distroless/static-debian11:nonroot` to `gcr.io/distroless/static-debian12:nonroot`

**Comment by markmandel** [comment]
@Kalaiselvi84 this would likely be good to for you to work on if you have run out of tasks 👍🏻

**Comment by Kalaiselvi84** [comment]
@markmandel, I've a few things to discuss:
1. I have updated most of the example images. Will push the new image to the production once the PRs are merged and then create a separate PR to update the rest of the files with new tag.

2. The [build](https://paste.googleplex.com/create/) is failing with `bookworm` for `cpp-simple` image. Log is too large to fit in gist.github.com, so I used paste instead.

3. Could you please let me know how to test the changes in the `/build` and `/cmd` directories?

**Comment by markmandel** [comment]
> Could you please let me know how to test the changes in the /build and /cmd directories?

The standard CI process will test those -- or you can do a `make build-images && make push && make install` locally and see if Agones is up and running.

**Comment by Kalaiselvi84** [comment]
The command `make build-images && make push && make install` is failing in both directories!

**Comment by markmandel** [comment]
Should be run from `/build` - I'll check the draft PR's you wrote, see what I can see 👍🏻

**Comment by markmandel** [comment]
> 2\. The [build](https://paste.googleplex.com/create/) is failing with `bookworm` for `cpp-simple` image. Log is too large to fit in gist.github.com, so I used paste instead.

Doesn't look like I got a gist / paste. Would likely need to see a draft PR and the output to help debug.

**Comment by markmandel** [comment]
Just a reminder that once #3576 is merged and published, will need to replace all the `simple-game-server:0.23` instances with `simple-game-server:0.24` with a follow up PR (e2e tests, docs, etc)

**Comment by markmandel** [comment]
Another reminder once #3575 is is merged and published, will need to replace all the autoscaler-webhook:0.9 instances with autoscaler-webhook:0.10 with a follow up PR (e2e tests, docs, etc),

**Comment by Kalaiselvi84** [comment]
@markmandel, We have bullseye in build/build-image/Dockerfile - https://github.com/googleforgames/agones/blob/589f77b4bddac84b6ab225cb62b44aa181b53cbd/build/build-image/Dockerfile#L17

Are we not upgrading it? 🤔Thought your PR has the fix for the `google-cloud-cli-app-engine-python : Depends: python2.7 but it is not installable` issue

**Comment by markmandel** [comment]
No I didn't, I just had the sdk updates. I was figuring we were blocked on https://b.corp.google.com/issues/265528492

But if you have a workaround, please go ahead and get it implemented 👍🏻
